### ISSUE TITLE
Featured image and sidebar images not CDNified

### ISSUE THREAD

**Issue by scarlac** [imgix/imgix-wordpress#13]
Featured images and images used in widgets from the site are not served through the imgix CDN since the plugin only attaches to the "the_content" filter which does not include anything but post content.

I'd argue that it should, by default, replace images in widgets as well as content. If not by default then at least offer it as an option.

**Comment by wladston** [comment]
@scarlac I agree with you. In this commit specifically

https://github.com/wladston/imgix-wordpress/commit/71dea46bdb018f06bf1b09d60cd45204152e8a3b

I'm hooking the retina feature to ALL images. I think the hook to replace images should be the same one as the one used for the retina hook. Maybe I could additionally also hook the imgix image source replacement to this all-HTML encompassing rule, what do you think?

**Comment by scarlac** [comment]
I'm not sure, @wladston . Perhaps adding a filter on `wp_get_attachment_image_src` or similar is more "the wordpress way", although that wouldn't catch references in widgets. If not, then I agree - it should search-replace on pretty much all content for maximum flexibility.

**Comment by wladston** [comment]
@scarlac yeah, that why I didn't do it yet. I already did the filter on `image_downsize` (https://github.com/wladston/imgix-wordpress/commit/a6760a536a2f6f6dbf34d12ca04708249e0b90dc), which should take care of everything handled by Wordpress.

**Comment by scarlac** [comment]
Yeah. This plugin should just be able to disable thumbnail generation alltogether and replace source images with imgix url. That'd speed up both uploading/editing as well as reader experience considerably.

**Comment by kellysutton** [comment]
@scarlac @wladston You folks are definitely correct. Once we merge in https://github.com/imgix/imgix-wordpress/pull/12, we should look at what it will take to get what @scarlac describes done.

**Comment by wladston** [comment]
Using #12 already requires absolutely nothing of Wordpress image resizing. If my guess is correct, once #12 is applied, Wordpress will stop resizing images altogether.

**Comment by kellysutton** [comment]
Just merged in @wladston's work and released 1.1.0. Can you pull that down and let me know if that fixes your issue?

**Comment by Glark** [comment]
Hi, I just applied last week's commit and I'm not seeing CDNification of feature images nor widget images. I can't quite follow the convo here so I'm not 100% I should be seeing them yet (not a programmer, obvs).

**Comment by Glark** [comment]
Would a imgix-controlled WP also "imgix-ify" thumbnails in the admin? There's lots of places they are used there (featured image preview, media insert modal, etc). Thought I'd mention it.

**Comment by scarlac** [comment]
@Glark Yes, I believe so. On my test environment it certainly points to imgix for the Media library default list. But it's not impossible that it'll point to "native" paths in some places outside of `image_downsize`'s jurisdiction. I'm still testing some changes on my own fork here.

**Comment by scarlac** [comment]
@Glark It's not replaced everywhere. the "File URL" field when editing a media file will display the native path, not imgix. But the image will itself be loaded from imgix. One could wonder why there's a difference but there is:
![screen shot 2015-10-14 at 14 19 59](https://cloud.githubusercontent.com/assets/895369/10483260/bec7b280-727e-11e5-8852-4f72fd4adc79.png)

to fix this you'd have to filter wp_upload_dir (upload_dir) and possibly other functions as well.
